libs/framework/src/framework_bundle_lifecycle_handler.c

```diff
@@ -83,31 +90,32 @@ void celix_framework_cleanupBundleLifecycleHandlers(celix_framework_t* fw, bool
     celixThreadMutex_unlock(&fw->bundleLifecycleHandling.mutex);
 }
 
-static void celix_framework_createAndStartBundleLifecycleHandler(celix_framework_t* fw, celix_framework_bundle_entry_t* bndEntry, enum celix_bundle_lifecycle_command bndCommand) {
+static void celix_framework_createAndStartBundleLifecycleHandler(celix_framework_t* fw, celix_framework_bundle_entry_t* bndEntry, enum celix_bundle_lifecycle_command bndCommand, const char* updatedBundleUrl) {
     celix_framework_bundleEntry_increaseUseCount(bndEntry);
     celixThreadMutex_lock(&fw->bundleLifecycleHandling.mutex);
     celix_framework_bundle_lifecycle_handler_t* handler = calloc(1, sizeof(*handler));
     handler->command = bndCommand;
     handler->framework = fw;
     handler->bndEntry = bndEntry;
     handler->bndId = bndEntry->bndId;
+    handler->updatedBundleUrl = celix_utils_strdup(updatedBundleUrl);
     celix_arrayList_add(fw->bundleLifecycleHandling.bundleLifecycleHandlers, handler);
 
     fw_log(fw->logger, CELIX_LOG_LEVEL_TRACE, "Creating thread for %s bundle %li",
            celix_bundleLifecycleCommand_getDesc(handler->command) , handler->bndId);
-    celixThread_create(&handler->thread, NULL, celix_framework_stopStartBundleThread, handler);
+    celixThread_create(&handler->thread, NULL, celix_framework_BundleLifecycleHandlingThread, handler);
```

Comment:
Pthread is joinable by default. Without a `pthread_join`, resource leak happens. 
See `man 7 pthread_detach` for remedies.

Reply:
A commit is submitted to cleanup handlers as soon as they finish their jobs.

Reply:
That could indeed be an issue. 

Note that the bundle to stop and uninstall are collected during framework shutdown:
https://github.com/apache/celix/blob/feature/refactor_bundle_cache/libs/framework/src/framework.c#L1201-L1214

I think what is missing is to also then mark the framework as shutting down and not allowing any new bundle changes (install, stopping, starting, etc).

But IMO an improvement for another PR

Reply:
Yes, it should be done that way. 
Hopefully when we have done everything in the right way, mysterious issues like #482 will disappear like magic.